lib/tasks/pipeline.js

```diff
@@ -78,26 +101,40 @@ module.exports = Task.extend({
     return Promise.resolve(deployConfigFn(this.deployTarget));
   },
 
-  _registerPipelineHooks: function(addon, deployConfig) {
-    var self = this;
-    var configuredPluginList = deployConfig.plugins;
+  _aliasHash: function(deployConfig, plugins) {
+    var aliasList = deployConfig.plugins;
+    if(!aliasList) {
+      // no whitelist available, autodiscover all the plugins
+      aliasList = Object.keys(plugins);
+    }
 
-    if (this._isDeployPluginPack(addon)) {
-      addon.addons.forEach(function(nestedAddon){
-        self._registerPipelineHooks(nestedAddon, deployConfig);
-      });
-      return;
+    var aliasHash = Object.create(null);
+    var configuredNameRegex = /(.*?):(.*)/;
+
+    for(var i=0, len=aliasList.length; i<len; i++) {
+      var alias = aliasList[i];
+      var split = alias.split(':', 2);
+      var pluginName = split[0];
+      var configuredName = split[1] ? alias.match(configuredNameRegex)[2] : split[0];
+
+      aliasHash[configuredName] = plugins[pluginName];
+
+      if(!aliasHash[configuredName]) {
```

Comment:
If we're going to reject because there is a whitelist entry for a plugin that we don't know about, would it be a good idea to maybe collect a list of all the unknown plugins? instead of bailing out at the first instance, would it be better to be able to report to the user that there are "x, y and z" plugins that we don't know about?